src/db_parser/parser.py

```diff
@@ -0,0 +1,230 @@
+        # return self.key_value_pair()
+        return Field(*self.key_value_pair(), has_macro=has_macro)
+
+    def info(self):
+        """
+        Handler for an EPICS DB info field.
+        Example:
+             info(alarm, "SIMPLE_01")
+        Returns:
+            tuple of (key, value)
+        """
+        self.consume(TokenTypes.INFO)
+        # return self.key_value_pair()
+        return Field(*self.key_value_pair())
+
+    def alias_field(self):
+        """
+        Handler for an EPICS alias within a DB record
+        Example:
+            alias("$(P)RECORD1")
+        Returns:
+             value of the alias
+        """
+        self.consume(TokenTypes.ALIAS)
+        with self.bracket_delimited_block():
+            return self.value()
+
+    def record(self):
+        """
+        Handler for an EPICS DB record.
+        Example:
+            record(ai, "$(P)RECORDNAME") {
+                field(PINI, "YES")
+                field(VAL, "0")
+                info(alarm, "SIMPLE_01")
+                alias("$(P)ALIASRECORDNAME")
+            }
+        Returns:
+            dict:
+                "type": record type, e.g. "ai"
+                "name": record name, e.g. "$(P)RECORDNAME"
+                "fields": list of fields. Each item in the list is a (key, value) tuple
+                "infos": list of info fields. Each item in the list is a (key, value) tuple
+                "aliases": list of record names aliased to this record
+        """
+        fields = []
+        infos = []
+        aliases = []
+        line_has_macro = False
+        self.consume(TokenTypes.RECORD)
+        record_type, record_name = self.key_value_pair()
+
+        # Handle Macro before opening brace
+        if self.current_token.type == TokenTypes.MACRO:
+            self.consume(TokenTypes.MACRO)
+        # Special case for records with no body
+        if self.current_token.type != TokenTypes.L_BRACE:
+            # return {"type": record_type, "name": record_name, "fields": [], "infos": [], "aliases": []}
+            return Record(record_type, record_name, [], [], [])
+        with self.brace_delimited_block():
+            while self.current_token.type != TokenTypes.R_BRACE:
+                if self.current_token.type == TokenTypes.FIELD:
+                    fields.append(self.field(line_has_macro))
+                    line_has_macro = False
+                elif self.current_token.type == TokenTypes.INFO:
+                    infos.append(self.info())
+                    line_has_macro = False
+                elif self.current_token.type == TokenTypes.ALIAS:
+                    aliases.append(self.alias_field())
+                    line_has_macro = False
+                elif self.current_token.type == TokenTypes.MACRO:
+                    line_has_macro = True
+                    self.consume(TokenTypes.MACRO)
+                else:
+                    self.raise_error("Expected info, field or alias")
```

Comment:
Bit cleaner:
```suggestion
                line_has_macro = False
                if self.current_token.type == TokenTypes.FIELD:
                    fields.append(self.field(line_has_macro))
                elif self.current_token.type == TokenTypes.INFO:
                    infos.append(self.info())
                elif self.current_token.type == TokenTypes.ALIAS:
                    aliases.append(self.alias_field())
                elif self.current_token.type == TokenTypes.MACRO:
                    line_has_macro = True
                    self.consume(TokenTypes.MACRO)
                else:
                    self.raise_error("Expected info, field or alias")
```

Reply:
I believe this would change the behaviour, since it is now setting `line_has_macro` to false before apending the field with the current value of line has macro, whereas the original appends `line_has_macro`'s current value before updating it for the next line?

Reply:
Yh, I see. What about just adding `line_has_macro = self.current_token.type == TokenTypes.MACRO` at the end? I think you should rename the macro to something like `previous_token_macro` as it's more informative of what it's actually telling you.

Reply:
I don't think I can move it to the end? removing the elif would make it  fall over since it would go into the else at the end, and once self.consum() has been called the current token won't be a macro anymore?

Reply:
What do you think of:
```python
if self.current_token.type == TokenTypes.MACRO:
   previous_token_macro = True
   self.consume(TokenTypes.MACRO)
   continue
if self.current_token.type == TokenTypes.FIELD:
    fields.append(self.field(previous_token_macro))
elif self.current_token.type == TokenTypes.INFO:
    infos.append(self.info())
elif self.current_token.type == TokenTypes.ALIAS:
    aliases.append(self.alias_field())
else:
    self.raise_error("Expected info, field or alias")
previous_token_macro = False
```

I'm not sure if it's more complicated or not, I like that it deals with the MACRO case first and gets it out of the way though.